src/main/java/dev/revere/alley/base/server/ServerService.java

```diff
@@ -29,18 +39,40 @@ public class ServerService implements IServerService {
     private final IProfileService profileService;
     private final IHotbarService hotbarService;
     private final ISpawnService spawnService;
+    private final IConfigService configService;
 
     private boolean queueingAllowed;
 
-    public ServerService(IMatchService matchService, IPartyService partyService, IProfileService profileService, IHotbarService hotbarService, ISpawnService spawnService) {
+    private List<Material> blockedCraftingItems;
+    private final String BLOCKED_MATERIALS_SECTION = "blocked-crafting-items";
+
+    /**
+     * Constructor for dependency injection.
+     *
+     * @param matchService   The match service.
+     * @param partyService   The party service.
+     * @param profileService The profile service.
+     * @param hotbarService  The hotbar service.
+     * @param spawnService   The spawn service.
+     * @param configService  The config service.
+     */
+    public ServerService(IMatchService matchService, IPartyService partyService, IProfileService profileService, IHotbarService hotbarService, ISpawnService spawnService, IConfigService configService) {
         this.matchService = matchService;
         this.partyService = partyService;
         this.profileService = profileService;
         this.hotbarService = hotbarService;
         this.spawnService = spawnService;
+        this.configService = configService;
+    }
+
+    @Override
+    public void initialize(AlleyContext context) {
         this.queueingAllowed = true;
+        this.blockedCraftingItems = new ArrayList<>();
+        this.loadBlockedCraftingMaterials();
```

Comment:
method is named loadBlockedCraftingMaterials but list is blockedCraftingItems?

Reply:
also just initialize blockedCraftingItems at declaration + remember to make it a Set. aka:


private final Set<Material> blockedCraftingItems = new HashSet<>();